plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/CopyPropertiesAction.java

```diff
@@ -0,0 +1,109 @@
+            // propType Check
+            PropertyType propertyType = profileService.getPropertyType(entry.getKey());
+            String propertyName = "properties." + entry.getKey();
+            if (propertyType != null && propertyType.isMultivalued()) {
+                changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValue");
+            } else {
+                Object profileProperty = event.getProfile().getProperty(entry.getKey());
+                if (profileProperty == null) {
+                    if (entry.getValue() instanceof List) {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValues");
+                    } else {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "alwaysSet");
+                    }
+                } else if (profileProperty instanceof List) {
+                    if (entry.getValue() instanceof List) {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValues");
+                    } else {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "addValue");
+                    }
+                } else {
+                    if (entry.getValue() instanceof List) {
+                        logger.warn("A single property named {} is already set on the profile.", entry.getKey());
+                    } else {
+                        changed = changed || PropertyHelper.setProperty(event.getProfile(), propertyName, entry.getValue(), "alwaysSet");
+                    }
+                }
+            }
+        }
+        return changed ? EventService.PROFILE_UPDATED : EventService.NO_CHANGE;
+    }
+
+    private Map<String, Object> getEventPropsToCopy(Action action, Event event) {
+        Map<String, Object> propsToCopy = new HashMap<String, Object>();
+
+        String rootProperty = (String) action.getParameterValues().get("rootProperty");
+        boolean copyEventProps = false;
+
+        if (StringUtils.isEmpty(rootProperty)) {
+            copyEventProps = true;
+            rootProperty = "target.properties";
+        }
+
+        // copy props from the event.properties
+        if (copyEventProps && event.getProperties() != null) {
+            propsToCopy.putAll(event.getProperties());
+        }
+
+        // copy props from the specified level (default is: target.properties)
+        try {
+            Object targetProperties = BeanUtilsBean.getInstance().getPropertyUtils().getProperty(event, rootProperty);
+            if (targetProperties instanceof Map) {
+                propsToCopy.putAll((Map) targetProperties);
+            }
+        } catch (Exception e) {
+            // Ignore
```

Comment:
could be cool to log something like: "Unable to extract properties to be copied from the event to the profile using root property: {}"

Reply:
Still required